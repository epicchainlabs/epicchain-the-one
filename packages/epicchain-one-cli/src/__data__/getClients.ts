import { getClients as getClientsBase } from '@neo-one/cli-common';
import { common, crypto } from '@neo-one/client-common';
import { NEOONEDataProvider } from '@neo-one/client-core';
import { constants } from '@neo-one/utils';

export const getClients = async (project: string) => {
  crypto.addPublicKey(
    common.stringToPrivateKey(constants.PRIVATE_NET_PRIVATE_KEY),
    common.stringToECPoint(constants.PRIVATE_NET_PUBLIC_KEY),
  );

  const config = await one.getProjectConfig(project);

  return getClientsBase(
    new NEOONEDataProvider({
      network: constants.LOCAL_NETWORK_NAME,
      rpcURL: `http://localhost:${config.network.port}/rpc`,
    }),
    constants.PRIVATE_NET_PRIVATE_KEY,
  );
};
