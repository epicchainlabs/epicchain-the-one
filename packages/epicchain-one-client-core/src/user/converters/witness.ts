import { Witness, WitnessModel } from '@neo-one/client-common';

export const witness = (witnessLike: Witness): WitnessModel =>
  new WitnessModel({
    invocation: Buffer.from(witnessLike.invocation, 'hex'),
    verification: Buffer.from(witnessLike.verification, 'hex'),
  });
