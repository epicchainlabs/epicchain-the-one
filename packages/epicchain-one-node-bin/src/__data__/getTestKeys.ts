import { common, crypto } from '@neo-one/client-common';
import { constants } from '@neo-one/utils';

export const getTestKeys = () => {
  const privateKey = common.stringToPrivateKey(constants.PRIVATE_NET_PRIVATE_KEY);
  const publicKey = common.stringToECPoint(constants.PRIVATE_NET_PUBLIC_KEY);
  crypto.addPublicKey(privateKey, publicKey);

  const standbyValidator = common.ecPointToString(publicKey);
  const address = common.uInt160ToString(crypto.privateKeyToScriptHash(privateKey));
  const privateKeyString = common.privateKeyToString(privateKey);

  return { standbyValidator, address, privateKeyString };
};
