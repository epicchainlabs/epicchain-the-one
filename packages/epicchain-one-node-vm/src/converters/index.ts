export * from './changes';
export * from './log';
export * from './stackItems';
