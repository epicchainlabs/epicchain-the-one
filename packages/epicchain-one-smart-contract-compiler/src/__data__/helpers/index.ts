export * from './compile';
export * from './createContext';
export * from './execute';
export * from './extractors';
export * from './startNode';
export * from './transpileAndExecute';
export * from './utils';
