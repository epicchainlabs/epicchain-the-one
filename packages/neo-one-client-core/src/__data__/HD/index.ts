export * from './seedOne';
export * from './seedThree';
export * from './seedTwo';
export * from './utils';
export * from './ledgerData';
