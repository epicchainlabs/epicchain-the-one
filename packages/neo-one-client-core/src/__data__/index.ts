export * from './keys';
export * from './data';
export * from './factory';
export * from './HD';
export * from './verifyDataProvider';
