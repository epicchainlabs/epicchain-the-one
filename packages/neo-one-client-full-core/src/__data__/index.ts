export * from './keys';
export * from './data';
export * from './factory';
