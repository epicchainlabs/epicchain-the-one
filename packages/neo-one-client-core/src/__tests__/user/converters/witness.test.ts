import { factory } from '../../../__data__';
import { witness } from '../../../user/converters/witness';

describe('witness', () => {
  test('converts to witness', () => {
    const value = factory.createWitness();

    const result = witness(value);

    expect(result.invocation).toMatchSnapshot();
    expect(result.verification).toMatchSnapshot();
  });
});
