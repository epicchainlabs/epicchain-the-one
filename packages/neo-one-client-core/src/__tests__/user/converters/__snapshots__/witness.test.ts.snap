// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`witness converts to witness 1`] = `"b500"`;

exports[`witness converts to witness 2`] = `"5f8d70"`;
